ports/data_parser.py

```diff
@@ -0,0 +1,30 @@
+class EmptyFileError(Exception):
```

Comment:
It is a good practice to have one, base (root) exception for all business exceptions raised by your application. For example:

```
class StudentGradesAppError(Exception):
    pass
```

and then all other exceptions derive from the base exception instead of `Exception`, e.g.:

```
class EmptyFileError(StudentGradesAppError):
    pass
```

This way there is possibility to have:
```
try:
    ...
except StudentGradesAppError:
   ...

```
to handle all custom exceptions raised by the application in one place without a need to list them all and remember to maintain this list in case of new exceptions added.